test/metabase/api/search_test.clj

```diff
@@ -724,3 +725,12 @@
         ;; we have this test here just to keep tracks this number to remind us to put effort
         ;; into keep this number as low as we can
         (is (= 7 (call-count)))))))
+
+(deftest snowplow-new-search-query-event-test
+  (testing "Send a snowplow event when a new search query is made"
```

Comment:
should we include a test "do not send a snowplow event if archived=true, nil? models, nil? table_id"?

Reply:
Added tests [Test for only tracking global searches](https://github.com/metabase/metabase/pull/31467/commits/fb06f1ba33bbad2b0dec194f876aaddf224c6330)